src/tunacode/tools/discover.py

```
@@ -1,31 +1,89 @@
+    args: JsonObject,
+    signal: asyncio.Event | None,
+    on_update: AgentToolUpdateCallback,
+) -> AgentToolResult:
+    _ = (tool_call_id, on_update)
+    if signal is not None and signal.is_set():
+        raise UserAbortError("Tool execution aborted: discover")
+
+    query = args.get("query")
+    directory = args.get("directory", ".")
+    if not isinstance(query, str):
+        raise ToolRetryError("Invalid arguments for tool 'discover': 'query' must be a string.")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)